src/components/sbb-tag-group/sbb-tag-group.e2e.ts

```diff
@@ -1,588 +1,558 @@
-        await page.waitForChanges();
+        tag2.checked = false;
+        await element.updateComplete;
 
-        expect(tag2).not.toHaveAttribute('checked');
-        expect(await tag2.getProperty('checked')).toBe(false);
-        expect(inputSpy).not.toHaveReceivedEvent();
-        expect(changeSpy).not.toHaveReceivedEvent();
-        expect(await element.getProperty('value')).toEqual(null);
+        expect(tag2).not.to.have.attribute('checked');
+        expect(tag2.checked).to.be.equal(false);
+        expect(inputSpy.count).not.to.be.greaterThan(0);
+        expect(changeSpy.count).not.to.be.greaterThan(0);
+        expect(element.value).to.be.equal(null);
       });
 
       it('should not emit events by setting checked programmatically to false [attribute]', async () => {
-        const changeSpy = await page.spyOnEvent('change');
-        const inputSpy = await page.spyOnEvent('input');
-        const tag2 = await page.find('sbb-tag#sbb-tag-2');
+        const changeSpy = new EventSpy('change');
+        const inputSpy = new EventSpy('input');
+        const tag2 = document.querySelector('sbb-tag#sbb-tag-2') as SbbTag;
 
-        await tag2.removeAttribute('checked');
-        await page.waitForChanges();
+        tag2.removeAttribute('checked');
+        await element.updateComplete;
 
-        expect(tag2).not.toHaveAttribute('checked');
-        expect(await tag2.getProperty('checked')).toBe(false);
-        expect(inputSpy).not.toHaveReceivedEvent();
-        expect(changeSpy).not.toHaveReceivedEvent();
-        expect(await element.getProperty('value')).toEqual(null);
+        expect(tag2).not.to.have.attribute('checked');
+        expect(tag2.checked).to.be.equal(false);
+        expect(inputSpy.count).not.to.be.greaterThan(0);
+        expect(changeSpy.count).not.to.be.greaterThan(0);
+        expect(element.value).to.be.equal(null);
       });
 
       it('should select another tag manually and uncheck others', async () => {
-        const changeSpy = await page.spyOnEvent('change');
-        const inputSpy = await page.spyOnEvent('input');
-        const tag2 = await page.find('sbb-tag#sbb-tag-2');
-        const tag3 = await page.find('sbb-tag#sbb-tag-3');
-
-        await tag3.click();
-        await page.waitForChanges();
-
-        expect(tag3).toHaveAttribute('checked');
-        expect(tag2).not.toHaveAttribute('checked');
-        expect(await tag2.getProperty('checked')).toBe(false);
-        expect(await tag3.getProperty('checked')).toBe(true);
+        const changeSpy = new EventSpy('change');
+        const inputSpy = new EventSpy('input');
+        const tag2 = document.querySelector('sbb-tag#sbb-tag-2') as SbbTag;
+        const tag3 = document.querySelector('sbb-tag#sbb-tag-3') as SbbTag;
+
+        tag3.click();
+        await element.updateComplete;
+
+        expect(tag3).to.have.attribute('checked');
+        expect(tag3.checked).to.be.equal(true);
+        await element.updateComplete;
```

Comment:
```suggestion
```

Reply:
If I remove this, the test will fail...

Reply:
It's a similar problem as also discussed in radio button group. has it some relations, that they are child components?

Reply:
what if you just group all "await element update" together (instead between expectations)?

Reply:
-> we introduce new logic later